source/index.md

```diff
@@ -68,6 +68,28 @@ The following illustrate of what can be achieved with the Nix ecosystem:
 - Atomic upgrades and rollbacks.
 
 
+## Who is Nix for?
+
+Nix is a tool for software developers.
+
+You don't have to be a professional and you don't need formal education in informatics to greatly benefit from Nix.
+For our purposes here, you are a software developer if you manipulate character strings to make computers do exactly what you want.
```

Comment:
I think this sentence doesn't help narrow down the audience.

I'm coming from this with mostly an assumption (roughly reinforced by answers in the Nix community surveys): that there are primarily two types of audience that come to Nix/NixOS. The first audience is people who actually write code and build software (I'm not using the term "software developer" here to avoid confusion with the current version of the PR). The second audience is people who are Linux enthusiasts - they use a Linux OS, know how to perform certain tasks on the terminal and are usually a bit more tech-savvy, but they are "script kiddies" at best when it comes to actual understanding about how software works.

With this assumption in mind, "manipulate character strings to make computers do exactly what you want" can be applied to both types of audience, but as I interpreted it from the other sources you listed, the focus is only on the first type of audience, not the second.

```suggestion
For our purposes here, you are a software developer if you write or build code for computers.
```

I feel this suggestion is excluding some part of the audience that is intended to be included, but I am not sure about that. I made this suggestion to start from the other end: very narrow, and hoping that we can identify other audiences that we want to include and keep adjusting the sentence.

Reply:
Fully agreed that there's a producer <-> consumer spectrum, but I really didn't want to go down that road here. 

My way of phrasing it is certainly peculiar. It's more to accommodate a sentiment expressed in https://github.com/NixOS/nixos-homepage/issues/828: I want the community to be welcoming to nerds of all sorts, and especially beginners. If you had asked me 10 years ago, I would not have identified with your definition. Beginners don't *write* code. They use it, they configure existing software (manipulating what isn't or what we usually don't consider code), sometimes read code, and then maybe start changing that code. The cool thing about Nix is that, while certainly you'll end up writing code, in the beginning it doesn't really feel like it.

The other thing I was trying to capture is that what you really need is a particular mind set (and a lot of time to learn things) rather than a particular occupation (which "writing code" is) or qualification. We eventually meet at what ends up being programming. It's just that at the boundaries there are enough people who don't actually want to program - but still do because they have to deal with the fact that software sucks or doesn't do what they want. Natural scientists are a good example. All the anecdotes I know  amount to "we write code [that breaks] all the time but would rather do `$SCIENCE`".

All this clearly excludes people who won't use the shell or a plain text editor.

That was my take at "not too wide, not too narrow".

Reply:
> All this clearly excludes people who won't use a command line interface or a plain text editor.

This sounds much more straightforward, how about just mentioning this instead?

Reply:
```suggestion
For our purposes here, you are a software developer if you use a command line interface and a plain text editor.
```

You mean like this?

Reply:
I see what you were going for now. From the PR description, I thought you wanted to closely align with the marketing team's audience focus (which seems to be closer to the definition I suggested).

If the idea is to keep it broader, why not remove the term "software developer" completely? It's bound to cause confusion (probably similar to mine).

My suggestion would be to shorten this to something like: "Nix is a tool for people comfortable with a command line interface and a plain text editor."

Reply:
> Nix is a tool for people comfortable with a command line interface and a plain text editor.

Hm, I agree it's more straightforward and descriptive. But also completely lacks the part that I find most important, and that's about the spirit of it all, the "why does it matter". Why would you bother going through pages of dry text for hours on end? Your suggestion makes it sound like this is yet another command line tool. I'm getting flashbacks from the vision statement discussion...

Okay, which group of people would care about the following?

- Makes computers *actually* do (well, almost) exactly the same thing every time (it's cached `execve()` in a sandbox on immutable files)
- Allows expressing process execution and file system operations in terms of declarative programs
- Exposes and allows managing all the complexity of the open source software ecosystem

If not forced by necessity, probably it's perfectionist control-freaks. Maybe that's the uncomfortable truth that prevents us from saying it out loud. If there's a constructive variant of it, I'd still take that.